dsc/workflows/opencourseware.py

```diff
@@ -381,20 +381,31 @@ def reconcile_bitstreams_and_metadata(self) -> bool:
         return reconciled
 
     def item_metadata_iter(self) -> Iterator[dict[str, Any]]:
-        """Yield source metadata from metadata JSON file in the zip file.
+        """Yield item metadata from metadata JSON file in the zip file.
 
-        The item identifiers are retrieved from the filenames of the zip
+        If the zip file does not include a metadata JSON file (data.json),
+        this method yields a dict containing only the item identifier.
+        Otherwise, a dict containing the item identifier and transformed metadata
+        is yielded.
+
+        NOTE: Item identifiers are retrieved from the filenames of the zip
         files, which follow the naming format "<item_identifier>.zip".
         """
         s3_client = S3Client()
         for file in s3_client.files_iter(
             bucket=self.s3_bucket, prefix=self.batch_path, file_type=".zip"
         ):
+            try:
+                source_metadata = self._read_metadata_from_zip_file(file)
+                transformed_metadata = self.metadata_transformer.transform(
+                    source_metadata
+                )
```

Comment:
Could this transformation move outside of this `try/except` block, which looks to only capture `FileNotFoundError`?

Reply:
Good catch! Updating. :)